modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts

```diff
@@ -104,36 +79,25 @@ const applyToMultiple = (editor, cells: Node[], data: CellData) => {
 };
 
 const onSubmitCellForm = (editor: Editor, cells: HTMLTableCellElement[], api) => {
-  const data = api.getData();
+  const data: CellData = api.getData();
   api.close();
 
   editor.undoManager.transact(() => {
-    const applicator = cells.length === 1 ? applyToSingle : applyToMultiple;
-    applicator(editor, cells, data);
+    applyCellData(editor, cells, data);
     editor.focus();
   });
 };
 
 const open = (editor: Editor) => {
-  // these any types are cheating, but seem difficult to unwind
-  let cellElm, cells = [];
-
-  // Get selected cells or the current cell
-  cells = editor.dom.select('td[data-mce-selected],th[data-mce-selected]');
-  cellElm = editor.dom.getParent(editor.selection.getStart(), 'td,th');
-  if (!cells.length && cellElm) {
-    cells.push(cellElm);
-  }
-
-  cellElm = cellElm || cells[0];
+  const cells = TableSelection.getCellsFromSelection(editor);
```

Comment:
This code is not the same as what was previously happening... previously it'd use the cell returned by `cellElm = editor.dom.getParent(editor.selection.getStart(), 'td,th');`, however it no longer uses this and instead returns an empty array. This seems like a bug in the new `getCellsFromSelection` function.

Reply:
After looking into it more, it does actually do this in `Selection.ts`. That's pretty confusing, but that seems to be the common thing we're finding with this code :disappointed: